src/actions/query.js

```diff
@@ -299,7 +314,8 @@ export function executeQuery(componentId, executeWatchList = false, mustExecuteM
 		});
 
 		if (finalQuery.length) {
-			dispatch(msearch(finalQuery, orderOfQueries));
+			// in case of an internal component the analytics headers should not be included
+			dispatch(msearch(finalQuery, orderOfQueries, false, componentId.endsWith('__internal')));
```

Comment:
This is redundant because internal components never fire a query.

Reply:
This passes a parameter to the `msearch` function to tell if the query was executed due to an internal component. Related to above comment. If there's a better approach we could try it as well.

Reply:
I think instead of blacklisting internal components, it would be best if we whitelist search analytics headers application only for the Result and ReactiveComponents.

Reply:
But when filters change that should also apply the search headers. Currently its working with search filters as well. Updating the PR in a while for search-filters support. Lets review this condition then.

Reply:
Resolving because we need search headers for filters as well